ethereum/core/src/main/java/tech/pegasys/teku/core/signatures/DeletableSigner.java

```diff
@@ -0,0 +1,115 @@
+import tech.pegasys.teku.spec.datastructures.state.ForkInfo;
+
+public class DeletableSigner implements Signer {
+  private final Signer delegate;
+  private boolean deleted = false;
+  private final ReadWriteLock lock = new ReentrantReadWriteLock();
+  private final Lock readLock = lock.readLock();
+
+  public DeletableSigner(final Signer delegate) {
+    this.delegate = delegate;
+  }
+
+  @Override
+  public void disable() {
+    lock.writeLock().lock();
+    try {
+      deleted = true;
+    } finally {
+      lock.writeLock().unlock();
+    }
+  }
+
+  @Override
+  public SafeFuture<BLSSignature> createRandaoReveal(final UInt64 epoch, final ForkInfo forkInfo) {
+    return sign(() -> delegate.createRandaoReveal(epoch, forkInfo));
+  }
+
+  @Override
+  public SafeFuture<BLSSignature> signBlock(final BeaconBlock block, final ForkInfo forkInfo) {
+    return sign(() -> delegate.signBlock(block, forkInfo));
+  }
+
+  @Override
+  public SafeFuture<BLSSignature> signAttestationData(
+      final AttestationData attestationData, final ForkInfo forkInfo) {
+    return sign(() -> delegate.signAttestationData(attestationData, forkInfo));
+  }
+
+  @Override
+  public SafeFuture<BLSSignature> signAggregationSlot(final UInt64 slot, final ForkInfo forkInfo) {
+    return sign(() -> delegate.signAggregationSlot(slot, forkInfo));
+  }
+
+  @Override
+  public SafeFuture<BLSSignature> signAggregateAndProof(
+      final AggregateAndProof aggregateAndProof, final ForkInfo forkInfo) {
+    return sign(() -> delegate.signAggregateAndProof(aggregateAndProof, forkInfo));
+  }
+
+  @Override
+  public SafeFuture<BLSSignature> signVoluntaryExit(
+      final VoluntaryExit voluntaryExit, final ForkInfo forkInfo) {
+    return sign(() -> delegate.signVoluntaryExit(voluntaryExit, forkInfo));
+  }
+
+  @Override
+  public SafeFuture<BLSSignature> signSyncCommitteeMessage(
+      final UInt64 slot, final Bytes32 beaconBlockRoot, final ForkInfo forkInfo) {
+    return sign(() -> delegate.signSyncCommitteeMessage(slot, beaconBlockRoot, forkInfo));
+  }
+
+  @Override
+  public SafeFuture<BLSSignature> signSyncCommitteeSelectionProof(
+      final SyncAggregatorSelectionData selectionData, final ForkInfo forkInfo) {
+    return sign(() -> delegate.signSyncCommitteeSelectionProof(selectionData, forkInfo));
+  }
+
+  @Override
+  public SafeFuture<BLSSignature> signContributionAndProof(
+      final ContributionAndProof contributionAndProof, final ForkInfo forkInfo) {
+    return sign(() -> delegate.signContributionAndProof(contributionAndProof, forkInfo));
+  }
+
+  @Override
+  public boolean isLocal() {
+    return true;
+  }
+
+  private SafeFuture<BLSSignature> sign(Supplier<SafeFuture<BLSSignature>> supplier) {
+    readLock.lock();
+    final SafeFuture<BLSSignature> future =
+        deleted ? SafeFuture.failedFuture(new SignerNotActiveException()) : supplier.get();
```

Comment:
For absolute safety:
```suggestion
        deleted ? SafeFuture.failedFuture(new SignerNotActiveException()) : SafeFuture.of(supplier);
```

May need to make the supplier an `ExceptionThrowingSupplier` just to make the compiler happy.